modules/imgcodecs/src/grfmt_gif.cpp

```diff
@@ -341,20 +342,24 @@ bool GifDecoder::lzwDecode() {
 
             // output code
             // 1. renew the lzw extra table
-            if (code < colorTableSize) {
-                lzwExtraTable[lzwTableSize].suffix = (uchar)code;
-                lzwTableSize ++;
-                lzwExtraTable[lzwTableSize].prefix.clear();
-                lzwExtraTable[lzwTableSize].prefix.push_back((uchar)code);
-                lzwExtraTable[lzwTableSize].length = 2;
-            } else if (code <= lzwTableSize) {
-                lzwExtraTable[lzwTableSize].suffix = lzwExtraTable[code].prefix[0];
-                lzwTableSize ++;
-                lzwExtraTable[lzwTableSize].prefix = lzwExtraTable[code].prefix;
-                lzwExtraTable[lzwTableSize].prefix.push_back(lzwExtraTable[code].suffix);
-                lzwExtraTable[lzwTableSize].length = lzwExtraTable[code].length + 1;
-            } else {
-                return false;
+            //    * notice that if the lzw table size is full, 
+            //    * we should use the old table until a clear code is encountered
+            if (lzwTableSize < 1<<12) {
```

Comment:
The value of `1 << 12` used in the function can be assigned to a variable and used.

Reply:
Not variable, but constant.